# **The World Bank's International Debt Data**

<span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">It's not that we humans only take debts to manage our necessities. A country may also take debt to manage its economy. For example, infrastructure spending is one costly ingredient required for a country's citizens to lead comfortable lives.&nbsp;</span> [The World Bank](https://www.worldbank.org/en/home) <span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">&nbsp;is the organization that provides debt to countries.</span>

In this notebook, we are going to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. We are going to find the answers to questions like:

- What is the total amount of debt that is owed by the countries listed in the dataset?
- Which country owns the maximum amount of debt and what does that amount look like?
- What is the average amount of debt owed by countries across different debt indicators?

This is part of [datacamp](/Users/rafsanalmamun/Downloads/Data-Analysis_Data-Science_Projects/SQL%20Projects/WB%20International%20Debt%20Statistics/dataset/www.datacamp.com)'s "[Analyze International Debt Statistics](https://www.datacamp.com/projects/754)" project.

### **\*\* Use the IDS Database \*\***

In [2]:
USE IDS;
GO

Commands completed successfully.

Total execution time: 00:00:00.009

## 1\. Overview of the 'international\_debt' Table

<span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">Let's first&nbsp;</span> `SELECT` _all_ <span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">of the columns from the&nbsp;</span> `international_debt` <span style="color: rgba(0, 0, 0, 0.87); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;; background-color: rgb(255, 255, 255);">table. Also, we'll limit the output to the first ten rows to keep the output clean.</span>

In [4]:
SELECT TOP(10) * FROM international_debt;

(10 rows affected)

Total execution time: 00:00:00.026

country_name country_code indicator_name indicator_code debt Afghanistan AFG PPG, bilateral (AMT, current US$) DT.AMT.BLAT.CD 61739336.9 Afghanistan AFG Principal repayments on external debt, long-term (AMT, current US$) DT.AMT.DLXF.CD 100847181.9 Afghanistan AFG PPG, multilateral (AMT, current US$) DT.AMT.MLAT.CD 39107845 Afghanistan AFG PPG, official creditors (AMT, current US$) DT.AMT.OFFT.CD 100847181.9 Afghanistan AFG PPG, bilateral (DIS, current US$) DT.DIS.BLAT.CD 49114729.4 Afghanistan AFG Disbursements on external debt, long-term (DIS, current US$) DT.DIS.DLXF.CD 72894453.7 Afghanistan AFG PPG, multilateral (DIS, current US$) DT.DIS.MLAT.CD 23779724.3 Afghanistan AFG PPG, official creditors (DIS, current US$) DT.DIS.OFFT.CD 72894453.7 Afghanistan AFG PPG, bilateral (INT, current US$) DT.INT.BLAT.CD 39903620.1 Afghanistan AFG Interest payments on external debt, long-term (INT, current US$) DT.INT.DLXF.CD 53239440.1

## 2\. Finding the Number of Distinct Countries

From the first ten rows, we can see the amount of debt owed by _Afghanistan_ in the different debt indicators. But we do not know the number of different countries we have on the table. There are repetitions in the country names because a country is most likely to have debt in more than one debt indicator.

Without a count of unique countries, we will not be able to perform our statistical analyses holistically. In this section, we are going to extract the number of unique countries present in the table.

In [5]:
SELECT COUNT(DISTINCT country_name) AS total_num_countries 
FROM international_debt;

(1 row affected)

Total execution time: 00:00:00.052

total_num_countries 124

## 3\. Finding Out the Distinct Debt Indicators

We can see there are a total of 124 countries present on the table. As we saw in the first section, there is a column called `indicator_name` that briefly specifies the purpose of taking the debt. Just beside that column, there is another column called `indicator_code` which symbolizes the category of these debts. Knowing about these various debt indicators will help us to understand the areas in which a country can possibly be indebted to.

In [6]:
SELECT DISTINCT indicator_code AS debt_indicators
FROM international_debt
ORDER BY debt_indicators;

(25 rows affected)

Total execution time: 00:00:00.032

debt_indicators DT.AMT.BLAT.CD DT.AMT.DLXF.CD DT.AMT.DPNG.CD DT.AMT.MLAT.CD DT.AMT.OFFT.CD DT.AMT.PBND.CD DT.AMT.PCBK.CD DT.AMT.PROP.CD DT.AMT.PRVT.CD DT.DIS.BLAT.CD DT.DIS.DLXF.CD DT.DIS.MLAT.CD DT.DIS.OFFT.CD DT.DIS.PCBK.CD DT.DIS.PROP.CD DT.DIS.PRVT.CD DT.INT.BLAT.CD DT.INT.DLXF.CD DT.INT.DPNG.CD DT.INT.MLAT.CD DT.INT.OFFT.CD DT.INT.PBND.CD DT.INT.PCBK.CD DT.INT.PROP.CD DT.INT.PRVT.CD

## 4\. Totaling the Amount of Debt Owed by the Countries

As mentioned earlier, the financial debt of a particular country represents its economic state. But if we were to project this on an overall global scale, how will we approach it?

Let's switch gears from the debt indicators now and find out the total amount of debt (in USD) that is owed by the different countries. This will give us a sense of how the overall economy of the entire world is holding up.

In [8]:
SELECT SUM(debt) as total_debt_USD
FROM international_debt;

(1 row affected)

Total execution time: 00:00:00.018

total_debt_USD 3079734487675.79

## 5\. Country with the Highest Debt

"Human beings cannot comprehend very large or very small numbers. It would be useful for us to acknowledge that fact." - [Daniel Kahneman](https://en.wikipedia.org/wiki/Daniel_Kahneman). That is more than _3 million **million**_ USD, an amount which is really hard for us to fathom.

Now that we have the exact total of the amounts of debt owed by several countries, let's now find out the country that owns the highest amount of debt along with the amount. **Note** that this debt is the sum of different debts owed by a country across several categories. This will help to understand more about the country in terms of its socio-economic scenarios. We can also find out the category in which the country owns its highest debt. But we will leave that for now.

In [11]:
SELECT TOP(1) WITH TIES country_name, ROUND(SUM(debt/1000000000), 2) AS total_debt_Billion_USD
FROM international_debt
GROUP BY country_name
ORDER BY total_debt_Billion_USD DESC;

(1 row affected)

Total execution time: 00:00:00.019

country_name total_debt_Billion_USD China 285.79

## 6\. Average Amount of Debt Across Indicators

So, it was _China_. A more in-depth breakdown of China's debts can be found [here](https://datatopics.worldbank.org/debt/ids/country/CHN).

We now have a brief overview of the dataset and a few of its summary statistics. We already have an idea of the different debt indicators in which the countries owe their debts. We can dig even further to find out on an average how much debt a country owes? This will give us a better sense of the distribution of the amount of debt across different indicators.

In [13]:
SELECT indicator_code, indicator_name, ROUND(AVG(debt)/1000000000,2) as avg_debt_Billion_USD
FROM international_debt
GROUP BY indicator_code, indicator_name
ORDER BY avg_debt_Billion_USD DESC;

(25 rows affected)

Total execution time: 00:00:00.040

indicator_code indicator_name avg_debt_Billion_USD DT.AMT.DLXF.CD Principal repayments on external debt, long-term (AMT, current US$) 5.9 DT.AMT.DPNG.CD Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$) 5.16 DT.DIS.DLXF.CD Disbursements on external debt, long-term (DIS, current US$) 2.15 DT.DIS.OFFT.CD PPG, official creditors (DIS, current US$) 1.96 DT.AMT.PRVT.CD PPG, private creditors (AMT, current US$) 1.8 DT.INT.DLXF.CD Interest payments on external debt, long-term (INT, current US$) 1.64 DT.INT.DPNG.CD Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$) 1.22 DT.DIS.BLAT.CD PPG, bilateral (DIS, current US$) 1.22 DT.AMT.OFFT.CD PPG, official creditors (AMT, current US$) 1.19 DT.AMT.PBND.CD PPG, bonds (AMT, current US$) 1.08 DT.DIS.MLAT.CD PPG, multilateral (DIS, current US$) 0.84 DT.INT.PBND.CD PPG, bonds (INT, current US$) 0.8 DT.AMT.PROP.CD PPG, other private creditors (AMT, current US$) 0.75 DT.AMT.PCBK.CD PPG, commercial banks (AMT, current US$) 0.73 DT.INT.PRVT.CD PPG, private creditors (INT, current US$) 0.72 DT.AMT.BLAT.CD PPG, bilateral (AMT, current US$) 0.71 DT.AMT.MLAT.CD PPG, multilateral (AMT, current US$) 0.49 DT.DIS.PRVT.CD PPG, private creditors (DIS, current US$) 0.31 DT.INT.OFFT.CD PPG, official creditors (INT, current US$) 0.3 DT.DIS.PCBK.CD PPG, commercial banks (DIS, current US$) 0.29 DT.INT.PCBK.CD PPG, commercial banks (INT, current US$) 0.16 DT.INT.BLAT.CD PPG, bilateral (INT, current US$) 0.16 DT.INT.MLAT.CD PPG, multilateral (INT, current US$) 0.14 DT.DIS.PROP.CD PPG, other private creditors (DIS, current US$) 0.08 DT.INT.PROP.CD PPG, other private creditors (INT, current US$) 0.03

## 7\. The Highest Amount of Principal Repayments

We can see that the indicator `DT.AMT.DLXF.CD` tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital. More information about this category can be found [here](https://datacatalog.worldbank.org/principal-repayments-external-debt-long-term-amt-current-us-0).

An interesting observation in the above finding is that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.

We can investigate this a bit more so as to find out which countries owe the highest amount of debt in the category of long term debts (`DT.AMT.DLXF.CD`). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically.

In [20]:
SELECT TOP(5) WITH TIES country_name, ROUND(debt/1000000000,2) as principal_repayment_debt_Billion_USD
FROM international_debt
WHERE indicator_code = 'DT.AMT.DLXF.CD'
ORDER BY debt DESC;

(5 rows affected)

Total execution time: 00:00:00.017

country_name principal_repayment_debt_Billion_USD China 96.22 Brazil 90.04 Russian Federation 66.59 Turkey 51.56 South Asia 48.76

## 8\. The Most Common Debt Indicator

China has the highest amount of debt in the long-term debt (`DT.AMT.DLXF.CD`) category. This is verified by [The World Bank](https://data.worldbank.org/indicator/DT.AMT.DLXF.CD?end=2018&most_recent_value_desc=true). It is often a good idea to verify our analyses like this since it validates that our investigations are correct.

We saw that long-term debt is the topmost category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? Let's find that out.

In [29]:
SELECT TOP(1) WITH TIES indicator_code, COUNT(*) AS country_count
FROM international_debt
GROUP BY indicator_code 
ORDER BY country_count DESC;

(6 rows affected)

Total execution time: 00:00:00.033

indicator_code country_count DT.AMT.DLXF.CD 124 DT.AMT.MLAT.CD 124 DT.INT.OFFT.CD 124 DT.AMT.OFFT.CD 124 DT.INT.DLXF.CD 124 DT.INT.MLAT.CD 124

## 9\. Other Viable Debt Issues and Conclusion

There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator `DT.AMT.DLXF.CD` is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue. But that is not the end of the story, a part of the story rather.

Let's change tracks from `debt_indicator`s now and focus on the amount of debt again. Let's find out the maximum amount of debt across the indicators along with the respective country names. With this, we will be in a position to identify the other plausible economic issues a country might be going through. By the end of this section, we will have found out the debt indicators in which a country owes its highest debt.

In [39]:
WITH max_debt_country AS
(
    SELECT country_name, MAX(debt) as max_debt
    FROM international_debt
    GROUP BY country_name
)
SELECT i.country_name, i.indicator_code, ROUND(i.debt/1000000000,2) AS debt_Billion_USD
FROM international_debt i
INNER JOIN max_debt_country
ON i.country_name = max_debt_country.country_name
AND max_debt_country.max_debt = i.debt
ORDER BY debt_Billion_USD DESC;

(166 rows affected)

Total execution time: 00:00:00.027

country_name indicator_code debt_Billion_USD China DT.AMT.DLXF.CD 96.22 Brazil DT.AMT.DLXF.CD 90.04 Russian Federation DT.AMT.DLXF.CD 66.59 Turkey DT.AMT.DLXF.CD 51.56 South Asia DT.AMT.DLXF.CD 48.76 Least developed countries: UN classification DT.DIS.DLXF.CD 40.16 IDA only DT.DIS.DLXF.CD 34.53 India DT.AMT.DLXF.CD 31.92 Indonesia DT.AMT.DLXF.CD 30.92 Kazakhstan DT.AMT.DLXF.CD 27.48 Mexico DT.AMT.DLXF.CD 25.22 Cameroon DT.DIS.DLXF.CD 18.19 Romania DT.AMT.DLXF.CD 14.01 Colombia DT.AMT.DLXF.CD 11.99 Angola DT.AMT.DLXF.CD 11.07 Venezuela, RB DT.AMT.DLXF.CD 9.88 Egypt, Arab Rep. DT.AMT.DLXF.CD 9.69 Lebanon DT.AMT.DLXF.CD 9.51 South Africa DT.AMT.DLXF.CD 9.47 Bangladesh DT.DIS.DLXF.CD 9.05 Bangladesh DT.DIS.OFFT.CD 9.05 Vietnam DT.AMT.DLXF.CD 8.87 Pakistan DT.AMT.DLXF.CD 8.34 Ukraine DT.AMT.DLXF.CD 8.15 Philippines DT.AMT.DLXF.CD 6.48 Thailand DT.AMT.DLXF.CD 5.91 Peru DT.AMT.DLXF.CD 5.88 Sri Lanka DT.AMT.DLXF.CD 5.76 Morocco DT.AMT.DLXF.CD 5.15 Belarus DT.AMT.DLXF.CD 4.64 Kenya DT.DIS.DLXF.CD 4.27 Ecuador DT.AMT.DLXF.CD 3.62 Serbia DT.AMT.DLXF.CD 3.41 Mongolia DT.AMT.DLXF.CD 3.3 Georgia DT.INT.DLXF.CD 3.09 Nigeria DT.AMT.DLXF.CD 2.96 Guatemala DT.AMT.DLXF.CD 2.91 Papua New Guinea DT.AMT.DLXF.CD 2.84 Ethiopia DT.DIS.DLXF.CD 2.68 El Salvador DT.AMT.DLXF.CD 2.57 Tunisia DT.AMT.DLXF.CD 2.42 Bosnia and Herzegovina DT.AMT.DLXF.CD 2.11 Uzbekistan DT.AMT.DLXF.CD 2.1 Jordan DT.AMT.DLXF.CD 1.99 Dominican Republic DT.AMT.DLXF.CD 1.98 Bulgaria DT.AMT.DLXF.CD 1.88 Costa Rica DT.AMT.DLXF.CD 1.78 Mauritius DT.AMT.DLXF.CD 1.76 Ghana DT.AMT.DLXF.CD 1.69 Senegal DT.DIS.DLXF.CD 1.66 Lao PDR DT.AMT.DLXF.CD 1.61 Paraguay DT.AMT.DLXF.CD 1.53 Azerbaijan DT.AMT.DLXF.CD 1.51 Zambia DT.DIS.DLXF.CD 1.5 Bolivia DT.DIS.DLXF.CD 1.42 Tanzania DT.AMT.DLXF.CD 1.39 Uganda DT.DIS.DLXF.CD 1.36 Myanmar DT.DIS.DLXF.CD 1.25 Myanmar DT.DIS.OFFT.CD 1.25 Nicaragua DT.AMT.DLXF.CD 1.19 Cote d'Ivoire DT.DIS.DLXF.CD 1.08 Cote d'Ivoire DT.DIS.OFFT.CD 1.08 Macedonia, FYR DT.AMT.DLXF.CD 0.88 Nepal DT.DIS.DLXF.CD 0.85 Nepal DT.DIS.OFFT.CD 0.85 Armenia DT.AMT.DLXF.CD 0.85 Kyrgyz Republic DT.DIS.DLXF.CD 0.84 Kyrgyz Republic DT.DIS.OFFT.CD 0.84 Albania DT.AMT.DLXF.CD 0.79 Cambodia DT.AMT.DLXF.CD 0.78 Moldova DT.AMT.DLXF.CD 0.77 Mozambique DT.DIS.DLXF.CD 0.75 Sudan DT.DIS.DLXF.CD 0.7 Sudan DT.DIS.OFFT.CD 0.7 Jamaica DT.INT.DLXF.CD 0.66 Montenegro DT.AMT.DLXF.CD 0.65 Benin DT.DIS.DLXF.CD 0.65 Burkina Faso DT.DIS.DLXF.CD 0.62 Burkina Faso DT.DIS.OFFT.CD 0.62 Niger DT.DIS.DLXF.CD 0.61 Niger DT.DIS.OFFT.CD 0.61 Mali DT.DIS.DLXF.CD 0.6 Mali DT.DIS.OFFT.CD 0.6 Mauritania DT.DIS.DLXF.CD 0.59 Mauritania DT.DIS.OFFT.CD 0.59 Gabon DT.DIS.DLXF.CD 0.52 Tajikistan DT.AMT.DLXF.CD 0.47 Zimbabwe DT.AMT.DLXF.CD 0.46 Madagascar DT.DIS.DLXF.CD 0.45 Honduras DT.AMT.DLXF.CD 0.41 Yemen, Rep. DT.DIS.OFFT.CD 0.4 Yemen, Rep. DT.DIS.DLXF.CD 0.4 Congo, Dem. Rep. DT.DIS.OFFT.CD 0.37 Congo, Dem. Rep. DT.DIS.DLXF.CD 0.37 Rwanda DT.DIS.DLXF.CD 0.33 Rwanda DT.DIS.OFFT.CD 0.33 Maldives DT.DIS.DLXF.CD 0.32 Syrian Arab Republic DT.AMT.DLXF.CD 0.31 Syrian Arab Republic DT.AMT.OFFT.CD 0.31 Congo, Rep. DT.DIS.DLXF.CD 0.25 Congo, Rep. DT.DIS.OFFT.CD 0.25 Togo DT.DIS.DLXF.CD 0.24 Togo DT.DIS.OFFT.CD 0.24 Kosovo DT.AMT.DLXF.CD 0.23 Chad DT.DIS.DLXF.CD 0.21 Chad DT.DIS.OFFT.CD 0.21 Bhutan DT.AMT.DLXF.CD 0.21 Djibouti DT.DIS.OFFT.CD 0.18 Djibouti DT.DIS.DLXF.CD 0.18 Guinea DT.DIS.DLXF.CD 0.18 Guinea DT.DIS.OFFT.CD 0.18 Algeria DT.AMT.DLXF.CD 0.17 Liberia DT.DIS.DLXF.CD 0.15 Liberia DT.DIS.OFFT.CD 0.15 Botswana DT.AMT.DLXF.CD 0.14 Malawi DT.DIS.DLXF.CD 0.13 Malawi DT.DIS.OFFT.CD 0.13 Iran, Islamic Rep. DT.AMT.DLXF.CD 0.11 Guyana DT.AMT.DLXF.CD 0.11 Eswatini DT.DIS.DLXF.CD 0.11 Eswatini DT.DIS.OFFT.CD 0.11 Cabo Verde DT.DIS.DLXF.CD 0.11 Belize DT.AMT.DLXF.CD 0.11 Afghanistan DT.AMT.DLXF.CD 0.1 Afghanistan DT.AMT.OFFT.CD 0.1 Sierra Leone DT.DIS.DLXF.CD 0.1 Sierra Leone DT.DIS.OFFT.CD 0.1 Gambia, The DT.DIS.DLXF.CD 0.09 Gambia, The DT.DIS.OFFT.CD 0.09 Lesotho DT.DIS.DLXF.CD 0.08 Lesotho DT.DIS.OFFT.CD 0.08 Fiji DT.DIS.DLXF.CD 0.07